### ISSUE TITLE
Mac Compilation

### ISSUE THREAD

**Issue by ivancevich** [j3ffhubb/musikernel#2]
Hey @j3ffhubb 

First of all I'd like to congratulate and thank you for such a good job on MusiKernel. I've just discovered it a few days ago.

I wanted to know if you have plans or at least think if it's possible to compile for Mac OS X. Also, it'd be nice to know a bit about what are your plans for the future of MusiKernel (I noticed you're working on v2).

I'd love to collaborate on such a good project, since I love making music and programming (I'm working with Golang mostly nowadays).

Best,
JC

**Comment by j3ffhubb** [comment]
Hello,

Thank you for the kind words.  I was looking to resume work on the Mac port soon, I'd love some help :-)

Some work was done on a Mac port 6 months ago, but I never finished it (it ended when my boss at the time bought my Mac from me, although I should be getting another one soon).  Fortunately, most of the work I did on the Windows port is also applicable to the Mac port.  If you wanted to try compiling it, I recommend using Homebrew and creating a Homebrew Formula modeled after the Windows/MSYS2 PKGBUILD script [here](https://github.com/j3ffhubb/musikernel/blob/master/windows/pkgbuild.py).  It may already work, or if not, it's probably not much work to finish it.  Of course I'm happy to help any way I can.

MusiKernel's UI is written in Python3 and PyQt5, the audio engine is written in C.  Since Golang is mostly a mixture of C and Python, you should feel comfortable with the source code :-)

Plans for MusiKernel 2:

Planned (roughly in order of priority):
* Big automation improvements (mostly replacing what's there now)
* Internal MIDI routing
* External MIDI output devices (MIDI input devices are already supported)
* Support for outputting to more than 2 channels of an audio device
* Various plugin improvements
* Per-note automation (although I'm not sure if this one can be done for MIDI CCs without some huge changes to the engine and plugins, but pitchbend should be easy enough)
* Native Mac port
* Anything else that can be improved

Already Finished:
* Plugins are now in a plugin rack instead of floating windows:  http://i.imgur.com/dAK2ujM.png  (the mixer also received some improvements as part of this work)

**Comment by ivancevich** [comment]
@j3ffhubb sounds very cool. What about VST plugins? Do you have any plans to support them?

One thing I'd like to know about is regarding the GUI. I've been reading a couple of forums of other not-mainstream DAWs and all of them agreed on having a pro-looking GUI would attract much more users to use their DAWs. Do you have any plans on this? I'm not an interface designer, but I'd love to help with this topic too.

**Comment by j3ffhubb** [comment]
```
What about VST plugins? Do you have any plans to support them? 
```

It's possible, but it's not planned for the initial release of MusiKernel v2.  The VST SDK is not open source compatible, which does create some legal challenges, but the real challenge is properly supporting all of the popular plugins.  There are so many plugins that are buggy and require quirks in the VST host code, I'd literally have to download dozens or hundreds of them, and test them all thoroughly and figure out why they crash (and how to fix it).  If you look at the forums of newer hosts like Bitwig, Reaper and Studio One, they've had to add quirks for many VST plugins, and those are VST hosts with multiple full time developers.

I'm focused on improvements to the sequencer and internal plugins for MusiKernel v2, but if somebody else wanted to contribute VST support, I would be open to it.  However, it would need to be tested and confirmed stable against all of the popular VST plugins before it would be merged, it would be bad for MusiKernel's reputation for stability to implement buggy VST support (especially since VST support would probably attract a lot more users).  Implementing VST support could probably be done in less than a week, but making it work properly with all of the VST plugins would take much longer.

``` 
One thing I'd like to know about is regarding the GUI. I've been reading a couple 
of forums of other non-mainstream DAWs and all of them agreed on having a 
pro-looking GUI would attract much more users to use their DAWs 
```

But, it already looks better than Ableton :-D

I created the UI theme it has now, but I'm a developer and not a UI designer by trade, so it's already at the limit of what I can do.  I would be open to someone with better graphic design skills reworking the theme.  The UI themes are (mostly) Qt-CSS, you can see the default theme [HERE](https://github.com/j3ffhubb/musikernel/tree/master/src/pydaw/themes/default).

**Comment by ivancevich** [comment]
@j3ffhubb I understand the VST stuff. Have you considered using a framework such as [JUCE](http://www.juce.com/) for enabling VST support? I think VST support is essential for today's music production. I'm not saying MusiKernel is useless without it, but it would great having it.

Regarding the UI, I'm not saying it looks bad. But of course there's room for improvements :)

I'll try to start with the Mac compilation, and then see what can i tackle after.

**Comment by j3ffhubb** [comment]
```
 I think VST support is essential for today's music production. I'm not saying MusiKernel 
is useless without it, but it would great having it.
```

No worries, I'd like to have VST support too :-)

MusiKernel is still a very young project (younger than any of the open source DAWs, younger than even Bitwig) and I've been the only developer so far (but I hope many more developers join).  I focused on internal plugins because I can ensure that they are stable, when a VST plugin crashes, it crashes the host, and users tend to assume that it was the host's fault (running them in a separate process can avoid crashing the host, but it has it's own problems).

I would definitely consider using JUCE for VST hosting.  It probably doesn't make sense to start work on VST support before the automation work I'm doing for MusiKernel v2, because the way it interfaces with VST automation would have to be re-written.  If MusiKernel had more developers, then it would be much faster to add VST support, since we could just come up with a list of supported VST plugins, then divide and conquer the list in parallel.

```
I'll try to start with the Mac compilation, and then see what can i tackle after.
```

Awesome, let me know if I can help in any way :-)

**Comment by j3ffhubb** [comment]
I've been working on a more modern color theme that reflects the current "flat" styles that everybody is pushing, and a more standard color palette consistent with the major DAWs.  [This is what it looks like right now](http://imgur.com/a/VULTb), but it can still change between now and release.

**Comment by ivancevich** [comment]
@j3ffhubb it's looking much better. very nice! i really like these buttons:
![image](https://cloud.githubusercontent.com/assets/2357833/10981116/8c1a268e-83e4-11e5-9f0a-d833fd659309.png)

**Comment by j3ffhubb** [comment]
[Here is a more finished version of the new theme](http://imgur.com/a/lSzUD)

I think the new theme really is at the limit of my abilities right now, LOL, so that's probably about what the final release will look like, unless a graphic designer decides to join the project and continue work on the new theme.

**Comment by ivancevich** [comment]
@j3ffhubb I'll see if I can help with that, but first I need to be able to compile to Mac. I've been investigating a bit, but didn't have to time to try yet. Hopefully I can give it a try during the weekend.

**Comment by ivancevich** [comment]
@j3ffhubb I've installed `python3` and `pyqt5` via Homebrew, and then run `make`, these are the results:
```
➜  src git:(master) make
make deps
make mingw_deps
cd sbsms && ( ./configure || ./configure --enable-sse=no ) && make
checking build system type... i386-apple-darwin15.0.0
checking host system type... i386-apple-darwin15.0.0
checking target system type... i386-apple-darwin15.0.0
checking for gcc... gcc
checking for C compiler default output file name... a.out
checking whether the C compiler works... yes
checking whether we are cross compiling... no
checking for suffix of executables...
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking for a sed that does not truncate output... /usr/bin/sed
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for fgrep... /usr/bin/grep -F
checking for ld used by gcc... /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld
checking if the linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) is GNU ld... no
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm
checking the name lister (/usr/bin/nm) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 196608
checking whether the shell understands some XSI constructs... yes
checking whether the shell understands "+="... yes
checking for /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld option to reload object files... -r
checking for objdump... no
checking how to recognize dependent libraries... pass_all
checking for ar... ar
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/bin/nm output from gcc object... ok
checking for dsymutil... dsymutil
checking for nmedit... nmedit
checking for lipo... lipo
checking for otool... otool
checking for otool64... no
checking for -single_module linker flag... yes
checking for -exported_symbols_list linker flag... yes
checking how to run the C preprocessor... gcc -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
rm: conftest.dSYM: is a directory
checking if gcc supports -fno-rtti -fno-exceptions... yes
checking for gcc option to produce PIC... -fno-common -DPIC
checking if gcc PIC flag -fno-common -DPIC works... yes
checking if gcc static flag -static works... no
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) supports shared libraries... yes
checking dynamic linker characteristics... darwin15.0.0 dyld
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... no
checking whether to build static libraries... yes
checking for g++... g++
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking whether we are using the GNU C++ compiler... (cached) yes
checking whether g++ accepts -g... (cached) yes
checking how to run the C++ preprocessor... g++ -E
checking for ld used by g++... /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld
checking if the linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) is GNU ld... no
checking whether the g++ linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) supports shared libraries... yes
checking for g++ option to produce PIC... -fno-common -DPIC
checking if g++ PIC flag -fno-common -DPIC works... yes
checking if g++ static flag -static works... no
checking if g++ supports -c -o file.o... yes
checking if g++ supports -c -o file.o... (cached) yes
checking whether the g++ linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) supports shared libraries... yes
checking dynamic linker characteristics... darwin15.0.0 dyld
checking how to hardcode library paths into programs... immediate
checking how to run the C++ preprocessor... g++ -E
checking for C++ compiler vendor... gnu
checking for lrint... yes
checking for lrintf... yes
gnu XXX
checking whether C++ compiler accepts -msse... yes
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... ./install-sh -c -d
checking for gawk... no
checking for mawk... no
checking for nawk... no
checking for awk... awk
checking whether make sets $(MAKE)... yes
checking for style of include used by make... GNU
checking dependency style of gcc... gcc3
checking dependency style of g++... gcc3
checking whether to enable maintainer-specific portions of Makefiles... no
configure: creating ./config.status
config.status: creating Makefile
config.status: creating src/Makefile
config.status: creating libsbsms.spec
config.status: creating sbsms.pc
config.status: creating src/config.h
config.status: executing libtool commands
config.status: executing depfiles commands
Making all in src
/Applications/Xcode.app/Contents/Developer/usr/bin/make  all-am
/bin/sh ../libtool --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I.    -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include  -MT sms.lo -MD -MP -MF .deps/sms.Tpo -c -o sms.lo sms.cpp
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include -MT sms.lo -MD -MP -MF .deps/sms.Tpo -c sms.cpp -o sms.o
mv -f .deps/sms.Tpo .deps/sms.Plo
/bin/sh ../libtool --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I.    -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include  -MT track.lo -MD -MP -MF .deps/track.Tpo -c -o track.lo track.cpp
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include -MT track.lo -MD -MP -MF .deps/track.Tpo -c track.cpp -o track.o
mv -f .deps/track.Tpo .deps/track.Plo
/bin/sh ../libtool --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I.    -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include  -MT trackpoint.lo -MD -MP -MF .deps/trackpoint.Tpo -c -o trackpoint.lo trackpoint.cpp
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include -MT trackpoint.lo -MD -MP -MF .deps/trackpoint.Tpo -c trackpoint.cpp -o trackpoint.o
mv -f .deps/trackpoint.Tpo .deps/trackpoint.Plo
/bin/sh ../libtool --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I.    -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include  -MT resample.lo -MD -MP -MF .deps/resample.Tpo -c -o resample.lo resample.cpp
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include -MT resample.lo -MD -MP -MF .deps/resample.Tpo -c resample.cpp -o resample.o
mv -f .deps/resample.Tpo .deps/resample.Plo
/bin/sh ../libtool --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I.    -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include  -MT sbsms.lo -MD -MP -MF .deps/sbsms.Tpo -c -o sbsms.lo sbsms.cpp
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include -MT sbsms.lo -MD -MP -MF .deps/sbsms.Tpo -c sbsms.cpp -o sbsms.o
mv -f .deps/sbsms.Tpo .deps/sbsms.Plo
/bin/sh ../libtool --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I.    -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include  -MT fft.lo -MD -MP -MF .deps/fft.Tpo -c -o fft.lo fft.cpp
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include -MT fft.lo -MD -MP -MF .deps/fft.Tpo -c fft.cpp -o fft.o
mv -f .deps/fft.Tpo .deps/fft.Plo
/bin/sh ../libtool --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I.    -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include  -MT grain.lo -MD -MP -MF .deps/grain.Tpo -c -o grain.lo grain.cpp
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include -MT grain.lo -MD -MP -MF .deps/grain.Tpo -c grain.cpp -o grain.o
mv -f .deps/grain.Tpo .deps/grain.Plo
/bin/sh ../libtool --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I.    -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include  -MT buffer.lo -MD -MP -MF .deps/buffer.Tpo -c -o buffer.lo buffer.cpp
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include -MT buffer.lo -MD -MP -MF .deps/buffer.Tpo -c buffer.cpp -o buffer.o
mv -f .deps/buffer.Tpo .deps/buffer.Plo
/bin/sh ../libtool --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I.    -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include  -MT subband.lo -MD -MP -MF .deps/subband.Tpo -c -o subband.lo subband.cpp
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include -MT subband.lo -MD -MP -MF .deps/subband.Tpo -c subband.cpp -o subband.o
mv -f .deps/subband.Tpo .deps/subband.Plo
/bin/sh ../libtool --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I.    -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include  -MT dBTable.lo -MD -MP -MF .deps/dBTable.Tpo -c -o dBTable.lo dBTable.cpp
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include -MT dBTable.lo -MD -MP -MF .deps/dBTable.Tpo -c dBTable.cpp -o dBTable.o
mv -f .deps/dBTable.Tpo .deps/dBTable.Plo
/bin/sh ../libtool --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I.    -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include  -MT slide.lo -MD -MP -MF .deps/slide.Tpo -c -o slide.lo slide.cpp
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include -MT slide.lo -MD -MP -MF .deps/slide.Tpo -c slide.cpp -o slide.o
mv -f .deps/slide.Tpo .deps/slide.Plo
/bin/sh ../libtool --tag=CXX   --mode=link g++ -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include    -o libsbsms.la -rpath /usr/local/lib sms.lo track.lo trackpoint.lo resample.lo sbsms.lo fft.lo grain.lo buffer.lo subband.lo dBTable.lo slide.lo
libtool: link: ar cru .libs/libsbsms.a  sms.o track.o trackpoint.o resample.o sbsms.o fft.o grain.o buffer.o subband.o dBTable.o slide.o
libtool: link: ranlib .libs/libsbsms.a
libtool: link: ( cd ".libs" && rm -f "libsbsms.la" && ln -s "../libsbsms.la" "libsbsms.la" )
make[4]: Nothing to be done for `all-am'.
cd sbsms/cli && make
make sbsms
c++ -c src/pcm.cpp -O3  -ffast-math -fprefetch-loop-arrays -floop-optimize -fsingle-precision-constant -funroll-loops -Wall -Isrc -I../include -fstrength-reduce -fstrict-aliasing -finline-functions -o src/pcm.o
clang: error: unknown argument: '-floop-optimize'
clang: warning: optimization flag '-fprefetch-loop-arrays' is not supported
clang: warning: optimization flag '-fsingle-precision-constant' is not supported
clang: warning: optimization flag '-fstrength-reduce' is not supported
clang: warning: optimization flag '-finline-functions' is not supported
make[4]: *** [src/pcm.o] Error 1
make[3]: *** [all] Error 2
make[2]: *** [mingw_deps] Error 2
make[1]: *** [deps] Error 2
make: *** [all] Error 2
```
I didn't have time to analyse those results, but wanted to share them with you.

**Comment by j3ffhubb** [comment]
Oh, yeah... That should be easy...  I created a separate make target for Clang because it doesn't support a bunch of the GCC flags or GNUC stuff, try this:

```
make clang
```

**Comment by j3ffhubb** [comment]
You also need more than pyqt5 and python3, the more complete list is something like:

```
"gcc-libs",  "rubberband", "portaudio", "portmidi", "libsndfile", "fftw", 
"python3-numpy", "libvorbis", "python3-pyqt5"
```

**Comment by ivancevich** [comment]
I keep getting this error `clang: error: unknown argument: '-floop-optimize'`.

Full logs:
```
➜  src git:(master) make clang
OPTIMIZE_FLAGS= CC=clang make
make deps
make mingw_deps
cd sbsms && ( ./configure || ./configure --enable-sse=no ) && make
checking build system type... i386-apple-darwin15.0.0
checking host system type... i386-apple-darwin15.0.0
checking target system type... i386-apple-darwin15.0.0
checking for gcc... clang
checking for C compiler default output file name... a.out
checking whether the C compiler works... yes
checking whether we are cross compiling... no
checking for suffix of executables...
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether clang accepts -g... yes
checking for clang option to accept ISO C89... none needed
checking for a sed that does not truncate output... /usr/bin/sed
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for fgrep... /usr/bin/grep -F
checking for ld used by clang... /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld
checking if the linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) is GNU ld... no
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm
checking the name lister (/usr/bin/nm) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 196608
checking whether the shell understands some XSI constructs... yes
checking whether the shell understands "+="... yes
checking for /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld option to reload object files... -r
checking for objdump... no
checking how to recognize dependent libraries... pass_all
checking for ar... ar
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/bin/nm output from clang object... ok
checking for dsymutil... dsymutil
checking for nmedit... nmedit
checking for lipo... lipo
checking for otool... otool
checking for otool64... no
checking for -single_module linker flag... yes
checking for -exported_symbols_list linker flag... yes
checking how to run the C preprocessor... clang -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
rm: conftest.dSYM: is a directory
checking if clang supports -fno-rtti -fno-exceptions... yes
checking for clang option to produce PIC... -fno-common -DPIC
checking if clang PIC flag -fno-common -DPIC works... yes
checking if clang static flag -static works... no
checking if clang supports -c -o file.o... yes
checking if clang supports -c -o file.o... (cached) yes
checking whether the clang linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) supports shared libraries... yes
checking dynamic linker characteristics... darwin15.0.0 dyld
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... no
checking whether to build static libraries... yes
checking for g++... g++
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking whether we are using the GNU C++ compiler... (cached) yes
checking whether g++ accepts -g... (cached) yes
checking how to run the C++ preprocessor... g++ -E
checking for ld used by g++... /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld
checking if the linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) is GNU ld... no
checking whether the g++ linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) supports shared libraries... yes
checking for g++ option to produce PIC... -fno-common -DPIC
checking if g++ PIC flag -fno-common -DPIC works... yes
checking if g++ static flag -static works... no
checking if g++ supports -c -o file.o... yes
checking if g++ supports -c -o file.o... (cached) yes
checking whether the g++ linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) supports shared libraries... yes
checking dynamic linker characteristics... darwin15.0.0 dyld
checking how to hardcode library paths into programs... immediate
checking how to run the C++ preprocessor... g++ -E
checking for C++ compiler vendor... gnu
checking for lrint... yes
checking for lrintf... yes
gnu XXX
checking whether C++ compiler accepts -msse... yes
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... ./install-sh -c -d
checking for gawk... no
checking for mawk... no
checking for nawk... no
checking for awk... awk
checking whether make sets $(MAKE)... yes
checking for style of include used by make... GNU
checking dependency style of clang... gcc3
checking dependency style of g++... gcc3
checking whether to enable maintainer-specific portions of Makefiles... no
configure: creating ./config.status
config.status: creating Makefile
config.status: creating src/Makefile
config.status: creating libsbsms.spec
config.status: creating sbsms.pc
config.status: creating src/config.h
config.status: executing libtool commands
config.status: executing depfiles commands
Making all in src
/Applications/Xcode.app/Contents/Developer/usr/bin/make  all-am
/bin/sh ../libtool --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I.    -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include  -MT sms.lo -MD -MP -MF .deps/sms.Tpo -c -o sms.lo sms.cpp
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include -MT sms.lo -MD -MP -MF .deps/sms.Tpo -c sms.cpp -o sms.o
mv -f .deps/sms.Tpo .deps/sms.Plo
/bin/sh ../libtool --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I.    -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include  -MT track.lo -MD -MP -MF .deps/track.Tpo -c -o track.lo track.cpp
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include -MT track.lo -MD -MP -MF .deps/track.Tpo -c track.cpp -o track.o
mv -f .deps/track.Tpo .deps/track.Plo
/bin/sh ../libtool --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I.    -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include  -MT trackpoint.lo -MD -MP -MF .deps/trackpoint.Tpo -c -o trackpoint.lo trackpoint.cpp
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include -MT trackpoint.lo -MD -MP -MF .deps/trackpoint.Tpo -c trackpoint.cpp -o trackpoint.o
mv -f .deps/trackpoint.Tpo .deps/trackpoint.Plo
/bin/sh ../libtool --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I.    -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include  -MT resample.lo -MD -MP -MF .deps/resample.Tpo -c -o resample.lo resample.cpp
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include -MT resample.lo -MD -MP -MF .deps/resample.Tpo -c resample.cpp -o resample.o
mv -f .deps/resample.Tpo .deps/resample.Plo
/bin/sh ../libtool --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I.    -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include  -MT sbsms.lo -MD -MP -MF .deps/sbsms.Tpo -c -o sbsms.lo sbsms.cpp
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include -MT sbsms.lo -MD -MP -MF .deps/sbsms.Tpo -c sbsms.cpp -o sbsms.o
mv -f .deps/sbsms.Tpo .deps/sbsms.Plo
/bin/sh ../libtool --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I.    -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include  -MT fft.lo -MD -MP -MF .deps/fft.Tpo -c -o fft.lo fft.cpp
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include -MT fft.lo -MD -MP -MF .deps/fft.Tpo -c fft.cpp -o fft.o
mv -f .deps/fft.Tpo .deps/fft.Plo
/bin/sh ../libtool --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I.    -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include  -MT grain.lo -MD -MP -MF .deps/grain.Tpo -c -o grain.lo grain.cpp
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include -MT grain.lo -MD -MP -MF .deps/grain.Tpo -c grain.cpp -o grain.o
mv -f .deps/grain.Tpo .deps/grain.Plo
/bin/sh ../libtool --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I.    -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include  -MT buffer.lo -MD -MP -MF .deps/buffer.Tpo -c -o buffer.lo buffer.cpp
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include -MT buffer.lo -MD -MP -MF .deps/buffer.Tpo -c buffer.cpp -o buffer.o
mv -f .deps/buffer.Tpo .deps/buffer.Plo
/bin/sh ../libtool --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I.    -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include  -MT subband.lo -MD -MP -MF .deps/subband.Tpo -c -o subband.lo subband.cpp
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include -MT subband.lo -MD -MP -MF .deps/subband.Tpo -c subband.cpp -o subband.o
mv -f .deps/subband.Tpo .deps/subband.Plo
/bin/sh ../libtool --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I.    -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include  -MT dBTable.lo -MD -MP -MF .deps/dBTable.Tpo -c -o dBTable.lo dBTable.cpp
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include -MT dBTable.lo -MD -MP -MF .deps/dBTable.Tpo -c dBTable.cpp -o dBTable.o
mv -f .deps/dBTable.Tpo .deps/dBTable.Plo
/bin/sh ../libtool --tag=CXX   --mode=compile g++ -DHAVE_CONFIG_H -I.    -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include  -MT slide.lo -MD -MP -MF .deps/slide.Tpo -c -o slide.lo slide.cpp
libtool: compile:  g++ -DHAVE_CONFIG_H -I. -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include -MT slide.lo -MD -MP -MF .deps/slide.Tpo -c slide.cpp -o slide.o
mv -f .deps/slide.Tpo .deps/slide.Plo
/bin/sh ../libtool --tag=CXX   --mode=link g++ -ffast-math -funroll-loops -O3 -fomit-frame-pointer -msse -I../include    -o libsbsms.la -rpath /usr/local/lib sms.lo track.lo trackpoint.lo resample.lo sbsms.lo fft.lo grain.lo buffer.lo subband.lo dBTable.lo slide.lo
libtool: link: ar cru .libs/libsbsms.a  sms.o track.o trackpoint.o resample.o sbsms.o fft.o grain.o buffer.o subband.o dBTable.o slide.o
libtool: link: ranlib .libs/libsbsms.a
libtool: link: ( cd ".libs" && rm -f "libsbsms.la" && ln -s "../libsbsms.la" "libsbsms.la" )
make[5]: Nothing to be done for `all-am'.
cd sbsms/cli && make
make sbsms
c++ -c src/pcm.cpp -O3  -ffast-math -fprefetch-loop-arrays -floop-optimize -fsingle-precision-constant -funroll-loops -Wall -Isrc -I../include -fstrength-reduce -fstrict-aliasing -finline-functions -o src/pcm.o
clang: error: unknown argument: '-floop-optimize'
clang: warning: optimization flag '-fprefetch-loop-arrays' is not supported
clang: warning: optimization flag '-fsingle-precision-constant' is not supported
clang: warning: optimization flag '-fstrength-reduce' is not supported
clang: warning: optimization flag '-finline-functions' is not supported
make[5]: *** [src/pcm.o] Error 1
make[4]: *** [all] Error 2
make[3]: *** [mingw_deps] Error 2
make[2]: *** [deps] Error 2
make[1]: *** [all] Error 2
make: *** [clang] Error 2
```

So, I commented out all the `-floop-optimize` usages. And these are the results:
```
➜  src git:(master) ✗ make clang
OPTIMIZE_FLAGS= CC=clang make
make deps
make mingw_deps
cd sbsms && ( ./configure || ./configure --enable-sse=no ) && make
checking build system type... i386-apple-darwin15.0.0
checking host system type... i386-apple-darwin15.0.0
checking target system type... i386-apple-darwin15.0.0
checking for gcc... clang
checking for C compiler default output file name... a.out
checking whether the C compiler works... yes
checking whether we are cross compiling... no
checking for suffix of executables...
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether clang accepts -g... yes
checking for clang option to accept ISO C89... none needed
checking for a sed that does not truncate output... /usr/bin/sed
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for fgrep... /usr/bin/grep -F
checking for ld used by clang... /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld
checking if the linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) is GNU ld... no
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/nm
checking the name lister (/usr/bin/nm) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 196608
checking whether the shell understands some XSI constructs... yes
checking whether the shell understands "+="... yes
checking for /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld option to reload object files... -r
checking for objdump... no
checking how to recognize dependent libraries... pass_all
checking for ar... ar
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /usr/bin/nm output from clang object... ok
checking for dsymutil... dsymutil
checking for nmedit... nmedit
checking for lipo... lipo
checking for otool... otool
checking for otool64... no
checking for -single_module linker flag... yes
checking for -exported_symbols_list linker flag... yes
checking how to run the C preprocessor... clang -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
rm: conftest.dSYM: is a directory
checking if clang supports -fno-rtti -fno-exceptions... yes
checking for clang option to produce PIC... -fno-common -DPIC
checking if clang PIC flag -fno-common -DPIC works... yes
checking if clang static flag -static works... no
checking if clang supports -c -o file.o... yes
checking if clang supports -c -o file.o... (cached) yes
checking whether the clang linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) supports shared libraries... yes
checking dynamic linker characteristics... darwin15.0.0 dyld
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... no
checking whether to build static libraries... yes
checking for g++... g++
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking whether we are using the GNU C++ compiler... (cached) yes
checking whether g++ accepts -g... (cached) yes
checking how to run the C++ preprocessor... g++ -E
checking for ld used by g++... /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld
checking if the linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) is GNU ld... no
checking whether the g++ linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) supports shared libraries... yes
checking for g++ option to produce PIC... -fno-common -DPIC
checking if g++ PIC flag -fno-common -DPIC works... yes
checking if g++ static flag -static works... no
checking if g++ supports -c -o file.o... yes
checking if g++ supports -c -o file.o... (cached) yes
checking whether the g++ linker (/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld) supports shared libraries... yes
checking dynamic linker characteristics... darwin15.0.0 dyld
checking how to hardcode library paths into programs... immediate
checking how to run the C++ preprocessor... g++ -E
checking for C++ compiler vendor... gnu
checking for lrint... yes
checking for lrintf... yes
gnu XXX
checking whether C++ compiler accepts -msse... yes
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... ./install-sh -c -d
checking for gawk... no
checking for mawk... no
checking for nawk... no
checking for awk... awk
checking whether make sets $(MAKE)... yes
checking for style of include used by make... GNU
checking dependency style of clang... gcc3
checking dependency style of g++... gcc3
checking whether to enable maintainer-specific portions of Makefiles... no
configure: creating ./config.status
config.status: creating Makefile
config.status: creating src/Makefile
config.status: creating libsbsms.spec
config.status: creating sbsms.pc
config.status: creating src/config.h
config.status: src/config.h is unchanged
config.status: executing libtool commands
config.status: executing depfiles commands
Making all in src
/Applications/Xcode.app/Contents/Developer/usr/bin/make  all-am
make[5]: Nothing to be done for `all-am'.
cd sbsms/cli && make
make sbsms
c++ -c src/pcm.cpp -O3  -ffast-math -fprefetch-loop-arrays -fsingle-precision-constant -funroll-loops -Wall -Isrc -I../include -fstrength-reduce -fstrict-aliasing -finline-functions -o src/pcm.o
clang: warning: optimization flag '-fprefetch-loop-arrays' is not supported
clang: warning: optimization flag '-fsingle-precision-constant' is not supported
clang: warning: optimization flag '-fstrength-reduce' is not supported
clang: warning: optimization flag '-finline-functions' is not supported
clang: warning: argument unused during compilation: '-fprefetch-loop-arrays'
clang: warning: argument unused during compilation: '-fsingle-precision-constant'
clang: warning: argument unused during compilation: '-fstrength-reduce'
clang: warning: argument unused during compilation: '-finline-functions'
c++ -c src/audiobuffer.cpp -O3  -ffast-math -fprefetch-loop-arrays -fsingle-precision-constant -funroll-loops -Wall -Isrc -I../include -fstrength-reduce -fstrict-aliasing -finline-functions -o src/audiobuffer.o
clang: warning: optimization flag '-fprefetch-loop-arrays' is not supported
clang: warning: optimization flag '-fsingle-precision-constant' is not supported
clang: warning: optimization flag '-fstrength-reduce' is not supported
clang: warning: optimization flag '-finline-functions' is not supported
clang: warning: argument unused during compilation: '-fprefetch-loop-arrays'
clang: warning: argument unused during compilation: '-fsingle-precision-constant'
clang: warning: argument unused during compilation: '-fstrength-reduce'
clang: warning: argument unused during compilation: '-finline-functions'
c++ -c src/convert.cpp -O3  -ffast-math -fprefetch-loop-arrays -fsingle-precision-constant -funroll-loops -Wall -Isrc -I../include -fstrength-reduce -fstrict-aliasing -finline-functions -o src/convert.o
clang: warning: optimization flag '-fprefetch-loop-arrays' is not supported
clang: warning: optimization flag '-fsingle-precision-constant' is not supported
clang: warning: optimization flag '-fstrength-reduce' is not supported
clang: warning: optimization flag '-finline-functions' is not supported
clang: warning: argument unused during compilation: '-fprefetch-loop-arrays'
clang: warning: argument unused during compilation: '-fsingle-precision-constant'
clang: warning: argument unused during compilation: '-fstrength-reduce'
clang: warning: argument unused during compilation: '-finline-functions'
src/convert.cpp:189:14: warning: unused variable 'lastPercent' [-Wunused-variable]
        long lastPercent=0;
             ^
src/convert.cpp:157:11: warning: unused variable 'preProgress' [-Wunused-variable]
    float preProgress = 0.0f;
          ^
src/convert.cpp:142:8: warning: unused variable 'bWrite' [-Wunused-variable]
  bool bWrite = false;
       ^
src/convert.cpp:141:8: warning: unused variable 'bRead' [-Wunused-variable]
  bool bRead = false;
       ^
4 warnings generated.
c++ -c src/import.cpp -O3  -ffast-math -fprefetch-loop-arrays -fsingle-precision-constant -funroll-loops -Wall -Isrc -I../include -fstrength-reduce -fstrict-aliasing -finline-functions -o src/import.o
clang: warning: optimization flag '-fprefetch-loop-arrays' is not supported
clang: warning: optimization flag '-fsingle-precision-constant' is not supported
clang: warning: optimization flag '-fstrength-reduce' is not supported
clang: warning: optimization flag '-finline-functions' is not supported
clang: warning: argument unused during compilation: '-fprefetch-loop-arrays'
clang: warning: argument unused during compilation: '-fsingle-precision-constant'
clang: warning: argument unused during compilation: '-fstrength-reduce'
clang: warning: argument unused during compilation: '-finline-functions'
c++ -c src/test.cpp -O3  -ffast-math -fprefetch-loop-arrays -fsingle-precision-constant -funroll-loops -Wall -Isrc -I../include -fstrength-reduce -fstrict-aliasing -finline-functions -o src/test.o
clang: warning: optimization flag '-fprefetch-loop-arrays' is not supported
clang: warning: optimization flag '-fsingle-precision-constant' is not supported
clang: warning: optimization flag '-fstrength-reduce' is not supported
clang: warning: optimization flag '-finline-functions' is not supported
clang: warning: argument unused during compilation: '-fprefetch-loop-arrays'
clang: warning: argument unused during compilation: '-fsingle-precision-constant'
clang: warning: argument unused during compilation: '-fstrength-reduce'
clang: warning: argument unused during compilation: '-finline-functions'
c++ src/pcm.o src/audiobuffer.o src/convert.o src/import.o src/test.o ../src/.libs/libsbsms.a -lpthread  -L/usr/local/Cellar/libsndfile/1.0.25/lib -lsndfile -lm -o sbsms
cd pydaw/pyliblo-0.9.2 && python3 setup.py build && cp build/lib*/liblo* ../python/libpydaw
running build
running build_ext
building 'liblo' extension
creating build
creating build/temp.macosx-10.11-x86_64-3.5
creating build/temp.macosx-10.11-x86_64-3.5/src
clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/usr/local/include -I/usr/local/opt/openssl/include -I/usr/local/opt/sqlite/include -I/usr/local/Cellar/python3/3.5.0/Frameworks/Python.framework/Versions/3.5/include/python3.5m -c src/liblo.c -o build/temp.macosx-10.11-x86_64-3.5/src/liblo.o -fno-strict-aliasing -Werror-implicit-function-declaration -Wfatal-errors
src/liblo.c:314:10: fatal error: 'lo/lo.h' file not found
#include "lo/lo.h"
         ^
1 error generated.
error: command 'clang' failed with exit status 1
make[2]: *** [deps] Error 1
make[1]: *** [all] Error 2
make: *** [clang] Error 2
```

Am I missing something?

**Comment by j3ffhubb** [comment]
I know what's wrong, the fix is pretty simple...  You can install "liblo" as a work-around, but since liblo isn't used in the Windows version, I'll probably also remove it as a dependency from the Mac version.

**Comment by ivancevich** [comment]
I'm almost there:
```
➜  bin git:(master) pwd
/usr/local/bin
➜  bin git:(master) musikernel2.py
sys.argv == ['/usr/local/bin/musikernel2.py']
/usr/local/lib/musikernel2/pydaw/python
ICON_PATH = '/usr/local/share/pixmaps/musikernel2.png'



install prefix:  /usr/local



BIN_PATH == None
Using stylesheet /usr/local/lib/musikernel2/themes/default/default.pytheme
locale: en_US
encoding: UTF-8
Exception while setting locale, falling back to English (hopefully):

'_' not defined by Python gettext module, setting to lambda x: x
Using /usr/local/lib/musikernel2/pydaw/mkengine/musikernel2.dylib
Traceback (most recent call last):
  File "/usr/local/bin/musikernel2.py", line 42, in <module>
    import musikernel
  File "/usr/local/lib/musikernel2/pydaw/python/musikernel.py", line 1423, in <module>
    MAIN_WINDOW.setup()
  File "/usr/local/lib/musikernel2/pydaw/python/musikernel.py", line 471, in setup
    pydaw_util.load_engine_lib(engine_lib_callback)
  File "/usr/local/lib/musikernel2/pydaw/python/libpydaw/pydaw_util.py", line 147, in load_engine_lib
    ENGINE_LIB = ctypes.CDLL(f_dll)
  File "/usr/local/Cellar/python3/3.5.0/Frameworks/Python.framework/Versions/3.5/lib/python3.5/ctypes/__init__.py", line 347, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: dlopen(/usr/local/lib/musikernel2/pydaw/mkengine/musikernel2.dylib, 6): image not found
```

**Comment by j3ffhubb** [comment]
Check is musikernel2.dylib is somewhere else with this `find /usr/local -name 'musikernel2.dylib'` and copy it if so.

If not, you can build it with and manually copy to the directory with something like:
```
cd [your source code directory]/musikernel/src/pydaw
CC=clang OPTIMIZE_FLAGS= make lib
cp musikernel2.dylib /usr/local/lib/musikernel2/pydaw/mkengine/
```

If that works, I'll add a `make mac_osx` target to the Makefile for Mac that does all of this stuff automatically.

**Comment by ivancevich** [comment]
I think there's an error compiling that file:
```
11 warnings generated.
Package alsa was not found in the pkg-config search path.
Perhaps you should add the directory containing `alsa.pc'
to the PKG_CONFIG_PATH environment variable
No package 'alsa' found
clang src/main.o -lpthread -lportmidi -lportaudio -shared  -L/usr/local/Cellar/libsndfile/1.0.25/lib -L/usr/local/Cellar/fftw/3.3.4_1/lib -lsndfile -lfftw3f -lm  -L/usr/local/Cellar/liblo/0.28/lib -llo -o musikernel2-engine
mv musikernel2-engine.exe musikernel2.dll || \
	mv musikernel2-engine musikernel2.so
mv: rename musikernel2-engine.exe to musikernel2.dll: No such file or directory
```
I wasn't able to find it neither before or after the `make lib` execution.

**Comment by j3ffhubb** [comment]
What does it say when you run `ls` in that directory?  I think it did work, but the file name was not what it expected it to be

**Comment by ivancevich** [comment]
```
➜  pydaw git:(master) ✗ pwd
/Users/jcivancevich/Documents/Developer/Personal/musikernel/src/pydaw
➜  pydaw git:(master) ✗ l
total 2400
drwxr-xr-x  21 jcivancevich  staff   714B Nov  8 01:32 .
drwxr-xr-x  16 jcivancevich  staff   544B Nov  8 00:36 ..
-rw-r--r--   1 jcivancevich  staff   4.9K Nov  8 00:43 Makefile
-rwxr-xr-x   1 jcivancevich  staff   108B Nov  8 00:36 i8n.sh
drwxr-xr-x   3 jcivancevich  staff   102B Nov  8 00:36 include
drwxr-xr-x   5 jcivancevich  staff   170B Nov  8 00:36 libmodsynth
drwxr-xr-x   6 jcivancevich  staff   204B Nov  8 00:36 locale
-rw-r--r--   1 jcivancevich  staff   185B Nov  8 00:36 mingw64-source-me.sh
-rw-r--r--   1 jcivancevich  staff    72K Nov  8 00:36 musikernel.po
-rwxr-xr-x   1 jcivancevich  staff   577K Nov  8 01:23 musikernel2-engine-dbg
-rwxr-xr-x   1 jcivancevich  staff   1.2K Nov  8 00:36 musikernel2.py
-rwxr-xr-x   1 jcivancevich  staff   515K Nov  8 01:32 musikernel2.so
drwxr-xr-x  18 jcivancevich  staff   612B Nov  8 00:36 plugins
drwxr-xr-x   6 jcivancevich  staff   204B Nov  8 00:36 presets
drwxr-xr-x  15 jcivancevich  staff   510B Nov  8 00:45 pyliblo-0.9.2
drwxr-xr-x  12 jcivancevich  staff   408B Nov  8 00:36 python
drwxr-xr-x   5 jcivancevich  staff   170B Nov  8 00:36 share
drwxr-xr-x  16 jcivancevich  staff   544B Nov  8 01:32 src
drwxr-xr-x   3 jcivancevich  staff   102B Nov  8 00:36 themes
-rwxr-xr-x   1 jcivancevich  staff   2.4K Nov  8 00:36 update_translations.py
-rwxr-xr-x   1 jcivancevich  staff   229B Nov  8 00:36 vm-cleaner
```

**Comment by j3ffhubb** [comment]
OK, I know what happened...  It renamed it to `musikernel2.so` instead of .dylib , so just do this:

```
mv musikernel2.so /usr/local/lib/musikernel2/pydaw/mkengine/musikernel2.dylib
```

**Comment by ivancevich** [comment]
<img width="1538" alt="screen shot 2015-11-08 at 01 36 06" src="https://cloud.githubusercontent.com/assets/2357833/11018794/1e6f8080-85b9-11e5-9ca7-1b1ec56fd21c.png">

:tada: :tada: :tada:

**Comment by j3ffhubb** [comment]
Awesome, you are literally the first person to ever get it working on OSX, I haven't even done that yet :)

Let me know if you find any bugs, and I'll attempt to fix them.  I'm about to merge a feature branch into master that has more UI updates and some big changes to automation.

I should be getting my Mac next week, so I can help test it then...

**Comment by ivancevich** [comment]
Sounds great! These are some changes I had to do to make it to work on Mac:

<img width="1529" alt="screen shot 2015-11-08 at 01 38 37" src="https://cloud.githubusercontent.com/assets/2357833/11018813/cc52df4e-85b9-11e5-9ec5-bc8b50a08b2a.png">

---

<img width="1528" alt="screen shot 2015-11-08 at 01 38 52" src="https://cloud.githubusercontent.com/assets/2357833/11018814/d7837b8a-85b9-11e5-89c2-71b357e0bac0.png">

---

<img width="1529" alt="screen shot 2015-11-08 at 01 39 31" src="https://cloud.githubusercontent.com/assets/2357833/11018815/e1287df2-85b9-11e5-9a34-e269948fc0ee.png">

---

<img width="1529" alt="screen shot 2015-11-08 at 01 39 46" src="https://cloud.githubusercontent.com/assets/2357833/11018816/eae30e66-85b9-11e5-9b03-f167f82c3e02.png">

---

<img width="1530" alt="screen shot 2015-11-08 at 01 39 57" src="https://cloud.githubusercontent.com/assets/2357833/11018818/f568112e-85b9-11e5-81aa-d456d9750249.png">

**Comment by ivancevich** [comment]
If you want, I can fork the repo and push those changes to a new branch, so you're able to see the diff on Github.

**Comment by j3ffhubb** [comment]
That would be awesome, please do :)

**Comment by ivancevich** [comment]
Done: https://github.com/ivancevich/musikernel/commit/9aa71a49183e48e3ab32156e6c5d6a9438e76cee

**Comment by j3ffhubb** [comment]
Thanks, I really appreciate you helping out with the Mac compiling and testing.  I'll create a `make mac_osx` target that does all of those things, and I'll post back here when it's done so you can test it.

**Comment by ivancevich** [comment]
Perfect :+1: I'm gonna test it and try to start diving into the code.

**Comment by j3ffhubb** [comment]
Here's the current instructions for building on Mac, let me know if they work:  https://github.com/j3ffhubb/musikernel/tree/master/mac_osx

**Comment by ivancevich** [comment]
These are the errors I get:
1) When running `make deps` I get `clang: error: unknown argument: '-floop-optimize'`. I removed all usages for that flag to be able to continue.

2) When running `make mac_osx` I get:
```
src/../include/../libmodsynth/lib/lmalloc.h:17:10: fatal error: 'malloc.h' file not found
#include <malloc.h>
         ^
1 error generated.
```
Which is solved by including it as `#include <malloc/malloc.h>`.

3) When running `PREFIX=/usr/local make install_non_linux` I get the following:
```
install -d /usr/local/lib/musikernel2
make -C pydaw PREFIX=/usr/local DESTDIR= install_non_linux
rm -rf python/*.pyc python/libpydaw/*.pyc python/*~ \
	python/libpydaw/*~ python/core python/*/__pycache__ \
	python/mkplugins/__pycache__
install -d /usr/local/bin
install -m 755 musikernel2.py /usr/local/bin
install -d /usr/local/lib/musikernel2
install -d /usr/local/lib/musikernel2/pydaw
cp -r python /usr/local/lib/musikernel2/pydaw
install -d /usr/local/lib/musikernel2/pydaw/mkengine
install -m 555 musikernel2.dll \
		/usr/local/lib/musikernel2/pydaw/mkengine || \
	install -m 555 musikernel2.so \
		/usr/local/lib/musikernel2/pydaw/mkengine || \
	install -m 555 musikernel2.dylib \
		/usr/local/lib/musikernel2/pydaw/mkengine
install: musikernel2.dll: No such file or directory
install: musikernel2.so: No such file or directory
cp -r themes /usr/local/lib/musikernel2
cp -r presets /usr/local/lib/musikernel2
cp -r share /usr/local
#locale/Makefile.py /usr/local
install -m 644 ../major-version.txt \
		/usr/local/lib/musikernel2
install -m 644 ../minor-version.txt \
		/usr/local/lib/musikernel2
#Add the Git revision to the version.txt file if possible
make -C pydaw_render DESTDIR= PREFIX=/usr/local install
install -d //usr/local
install -d //usr/local/bin
install -s -m 755 musikernel2_render //usr/local/bin
install: musikernel2_render: No such file or directory
make[1]: *** [install] Error 71
make: *** [install_non_linux] Error 2
```

**Comment by j3ffhubb** [comment]
I think that should fix all of it, thanks.  Please note that I changed the build  instructions to simply:

```
cd [source dir]/musikernel/src                                                  
make mac_osx                                                                    
make install_mac_osx 
```

**Comment by j3ffhubb** [comment]
Also, last night's automation merge broke a lot of stuff, but it should be all fixed in the latest master branch

**Comment by ivancevich** [comment]
@j3ffhubb it's still breaking with this error:
```
clang: error: unknown argument: '-floop-optimize'
clang: warning: optimization flag '-fprefetch-loop-arrays' is not supported
clang: warning: optimization flag '-fsingle-precision-constant' is not supported
clang: warning: optimization flag '-fstrength-reduce' is not supported
clang: warning: optimization flag '-finline-functions' is not supported
make[4]: *** [src/pcm.o] Error 1
make[3]: *** [all] Error 2
make[2]: *** [mingw_deps] Error 2
make[1]: *** [deps] Error 2
make: *** [mac_osx] Error 2
```

**Comment by j3ffhubb** [comment]
Sorry, I forgot about that Makefile.  It should be fixed now.

**Comment by ivancevich** [comment]
Cool. I'm working on the transport section, on the play, stop, rec buttons specifically.
<img width="801" alt="screen shot 2015-11-08 at 17 45 31" src="https://cloud.githubusercontent.com/assets/2357833/11022574/ad36074e-8640-11e5-8f92-7c384daeb75a.png">

**Comment by j3ffhubb** [comment]
Looks nice, thanks.

I will gladly accept your pull request whenever it's ready  :-)

**Comment by j3ffhubb** [comment]
Closing this issue, Mac OS X support should be pretty good as of the latest commits to the master branch.  I'll begin distributing the Mac OS X port using Homebrew or an installer package when MusiKernel 2 is officially released.

Any theme-related discussion can be moved to a pull request or a new issue.
